release-20.1: server: TestLogGCTrigger failed

[(server).TestLogGCTrigger failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2483927&tab=buildLog) on [release-20.1@9d57945825c83e38bac99781cb41091e51f2a2a9](https://github.com/cockroachdb/cockroach/commits/9d57945825c83e38bac99781cb41091e51f2a2a9):

```
github.com/cockroachdb/cockroach/pkg/kv/kvserver.init
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/node_liveness.go:57
runtime.doInit
	/usr/local/go/src/runtime/proc.go:5222
runtime.doInit
	/usr/local/go/src/runtime/proc.go:5217
runtime.doInit
	/usr/local/go/src/runtime/proc.go:5217
runtime.main
	/usr/local/go/src/runtime/proc.go:190
runtime.goexit
	/usr/local/go/src/runtime/asm_amd64.s:1357
I201201 06:13:57.441492 74853 server/status/runtime.go:516  [n1] runtime stats: 294 MiB RSS, 268 goroutines, 34 MiB/144 MiB/73 MiB GO alloc/idle/total, 21 MiB/78 MiB CGO alloc/total, 0.0 CGO/sec, 0.0/0.0 %(u/s)time, 0.0 %gc (125x), 25 MiB/25 MiB (r/w)net
I201201 06:14:07.655023 74853 server/status/runtime.go:516  [n1] runtime stats: 294 MiB RSS, 267 goroutines, 38 MiB/141 MiB/73 MiB GO alloc/idle/total, 21 MiB/78 MiB CGO alloc/total, 18.0 CGO/sec, 20.2/12.5 %(u/s)time, 0.0 %gc (0x), 49 MiB/49 MiB (r/w)net
I201201 06:14:07.656010 74861 kv/kvserver/node_liveness.go:821  [n1,liveness-hb] retrying liveness update after kvserver.errRetryLiveness: result is ambiguous (context done during DistSender.Send: context deadline exceeded)
W201201 06:14:07.656041 74861 kv/kvserver/node_liveness.go:563  [n1,liveness-hb] slow heartbeat took 15.7s
W201201 06:14:07.656081 74861 kv/kvserver/node_liveness.go:488  [n1,liveness-hb] failed node liveness heartbeat: operation "node liveness heartbeat" timed out after 4.5s
(1) operation "node liveness heartbeat" timed out after 4.5s
Wraps: (2) context deadline exceeded
Error types: (1) *contextutil.TimeoutError (2) context.deadlineExceededError
I201201 06:14:07.658013 74495 kv/kvserver/store.go:2474  [n1,s1] sstables (read amplification = 0):
I201201 06:14:07.658153 74495 kv/kvserver/store.go:2475  [n1,s1] 
** Compaction Stats [default] **
Level    Files   Size     Score Read(GB)  Rn(GB) Rnp1(GB) Write(GB) Wnew(GB) Moved(GB) W-Amp Rd(MB/s) Wr(MB/s) Comp(sec) CompMergeCPU(sec) Comp(cnt) Avg(sec) KeyIn KeyDrop
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Sum      0/0    0.00 KB   0.0      0.0     0.0      0.0       0.0      0.0       0.0   0.0      0.0      0.0      0.00              0.00         0    0.000       0      0
 Int      0/0    0.00 KB   0.0      0.0     0.0      0.0       0.0      0.0       0.0   0.0      0.0      0.0      0.00              0.00         0    0.000       0      0

** Compaction Stats [default] **
Priority    Files   Size     Score Read(GB)  Rn(GB) Rnp1(GB) Write(GB) Wnew(GB) Moved(GB) W-Amp Rd(MB/s) Wr(MB/s) Comp(sec) CompMergeCPU(sec) Comp(cnt) Avg(sec) KeyIn KeyDrop
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Uptime(secs): 20.3 total, 20.3 interval
Flush(GB): cumulative 0.000, interval 0.000
AddFile(GB): cumulative 0.000, interval 0.000
AddFile(Total Files): cumulative 0, interval 0
AddFile(L0 Files): cumulative 0, interval 0
AddFile(Keys): cumulative 0, interval 0
Cumulative compaction: 0.00 GB write, 0.00 MB/s write, 0.00 GB read, 0.00 MB/s read, 0.0 seconds
Interval compaction: 0.00 GB write, 0.00 MB/s write, 0.00 GB read, 0.00 MB/s read, 0.0 seconds
Stalls(count): 0 level0_slowdown, 0 level0_slowdown_with_compaction, 0 level0_numfiles, 0 level0_numfiles_with_compaction, 0 stop for pending_compaction_bytes, 0 slowdown for pending_compaction_bytes, 0 memtable_compaction, 0 memtable_slowdown, interval 0 total count
estimated_pending_compaction_bytes: 0 B
W201201 06:14:07.658396 75844 kv/kvserver/node_liveness.go:563  [n1,s1,r21/1:/Table/2{5-6}] slow heartbeat took 11.2s
    --- FAIL: TestLogGCTrigger/rangelog (20.03s)
        server_systemlog_gc_test.go:232: 
            	Error Trace:	server_systemlog_gc_test.go:232
            	Error:      	Not equal: 
            	            	expected: 0
            	            	actual  : 1
            	Test:       	TestLogGCTrigger/rangelog
            	Messages:   	Expected zero events before 2020-12-01 06:13:37.606736 +0000 +0000 after gc

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestLogGCTrigger PKG=./pkg/server TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestLogGCTrigger.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/server/server_systemlog_gc_test.go
+++ b/pkg/server/server_systemlog_gc_test.go
@@ -29,6 +29,7 @@ import (
 
 func TestLogGC(t *testing.T) {
 	defer leaktest.AfterTest(t)()
+	t.Skip("#57277") // Test flake; underlying cause fixed in v21.1.
 	a := assert.New(t)
 	s, db, _ := serverutils.StartServer(t, base.TestServerArgs{})
 	ts := s.(*TestServer)

